File: src/main/java/ee/carlrobert/codegpt/actions/toolwindow/MoveDownAction.java
package ee.carlrobert.codegpt.actions.toolwindow;

import com.intellij.icons.AllIcons;
import com.intellij.openapi.project.Project;
import ee.carlrobert.codegpt.actions.editor.EditorActionsUtil;
import ee.carlrobert.codegpt.conversations.Conversation;
import ee.carlrobert.codegpt.conversations.ConversationService;
import java.util.Optional;
import org.jetbrains.annotations.NotNull;

public class MoveDownAction extends MoveAction {

  public MoveDownAction(Runnable onRefresh) {
    super("Move Down", "Move Down", AllIcons.Actions.MoveDown, onRefresh);
    EditorActionsUtil.registerAction(this);
  }

  @Override
  protected Optional<Conversation> getConversation(@NotNull Project project) {
    return ConversationService.getInstance().getPreviousConversation();
  }
}

File: src/main/kotlin/ee/carlrobert/codegpt/actions/toolwindow/MoveDownAction.kt
package ee.carlrobert.codegpt.actions.toolwindow

import com.intellij.icons.AllIcons
import com.intellij.openapi.project.Project
import ee.carlrobert.codegpt.actions.editor.EditorActionsUtil.registerAction
import ee.carlrobert.codegpt.conversations.Conversation
import ee.carlrobert.codegpt.conversations.ConversationService
import java.util.Optional

class MoveDownAction(onRefresh: Runnable?) :
  MoveAction("Move Down", "Move Down", AllIcons.Actions.MoveDown, onRefresh!!) {
  init {
    registerAction(this)
  }

  override fun getConversation(project: Project): Optional<Conversation?> {
    return ConversationService.getInstance().previousConversation
  }
}
